modules/social_features/social_profile/src/Plugin/views/filter/SocialSearchApiSplitTerms.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\social_profile\Plugin\views\filter;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\search_api\Plugin\views\filter\SearchApiTerm;
+use Drupal\social_profile\SocialProfileTagServiceInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a filter for filtering on taxonomy term references.
+ *
+ * @ingroup views_filter_handlers
+ *
+ * @ViewsFilter("social_search_api_split_terms")
```

Comment:
@dmytriikaiun From the code, I see this filter works only for the profile tag field.
So maybe it's better to rename the machine name and class to something with a Social Profile prefix.